### PR TITLE
Change the way getRealMemUsage() works on Linux (using statm)

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

Use a less expensive way to get memory stats for rel-memory-usage. You can ask for the old way by explicitly asking for special-memory-usage.
Also, implement a getrusage() based getRealMemUsage() for non-Linux systems.

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [X] included documentation (including possible behaviour changes)
- [X] documented the code
- [X] added or modified regression test(s)
- [X] added or modified unit test(s)

### CODE REVIEW

File: pdns/dnsdistdist/DNSDIST-MIB.txt [hunk 1]
```diff
@@ -326,6 +326,14 @@ ruleServFail OBJECT-TYPE
 	"Number of ServFail responses returned because of a rule"
     ::= { stats 37 }
 
+specialMemoryUsage OBJECT-TYPE
```
Comment (rgacogne) [inline pdns/dnsdistdist/DNSDIST-MIB.txt]: This new object needs to be added to the dnsdistGroup OBJECT-GROUP.

File: pdns/recursordist/RECURSOR-MIB.txt [hunk 2]
```diff
@@ -809,6 +809,14 @@ variableResponses OBJECT-TYPE
         "Number of variable responses"
     ::= { stats 97 }
 
+specialMemoryUsage OBJECT-TYPE
```
Comment (rgacogne) [inline pdns/recursordist/RECURSOR-MIB.txt]: This new object needs to be added to the recGroup OBJECT-GROUP.

File: pdns/misc.cc [hunk 3]
```diff
@@ -1240,6 +1240,26 @@ uint64_t getOpenFileDescriptors(const std::string&)
 
 uint64_t getRealMemoryUsage(const std::string&)
 {
+#ifdef __linux__
+  ifstream ifs("/proc/"+std::to_string(getpid())+"/statm");
```
Comment (rgacogne) [inline pdns/misc.cc]:
```suggestion
  ifstream ifs("/proc/self/statm");
```

Comment (omoerbeek) [reply pdns/misc.cc]: Done

**Pull request merged**